nifi-nar-bundles/nifi-jms-bundle/nifi-jms-cf-service/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java

```diff
@@ -178,7 +178,7 @@ private void setConnectionFactoryProperties(ConfigurationContext context) {
             } else {
                 if (propertyName.equals(BROKER)) {
                     if (context.getProperty(CONNECTION_FACTORY_IMPL).evaluateAttributeExpressions().getValue().startsWith("org.apache.activemq")) {
-                        this.setProperty("brokerURL", entry.getValue());
+                        this.setProperty("brokerURL", context.getProperty(descriptor).evaluateAttributeExpressions().getValue());
```

Comment:
Unfortunately this only addresses the IF and not the ELSE where the value is treated differently to account for providers other then ActiveMQ. So, if we're adding EL support for  'brokerURL' we should do it for all. 

Also, with EL one must account for a possibility of null.

Anyway, if please address it when you get a chance or let us know if you don't have time and one of us can essentially collaborate and address the comments above.